src/main/java/seedu/address/logic/commands/delete/DeleteExamCommand.java

```diff
@@ -24,28 +23,27 @@
             + "\nExample: " + COMMAND_WORD + " exam 1 "
             + PREFIX_CODE + "CS2103T";
     private final Index targetIndex;
-    private final ModuleCode targetModuleCode;
 
     /**
      * Creates an DeleteExamCommand to delete the Exam at specified {@code Index} of the specified {@code Module}.
      */
-    public DeleteExamCommand(Index targetIndex, ModuleCode targetModuleCode) {
-        requireAllNonNull(targetIndex, targetModuleCode);
+    public DeleteExamCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
         this.targetIndex = targetIndex;
-        this.targetModuleCode = targetModuleCode;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        Module module = model.getModule(targetModuleCode);
+        Module module = model.getFilteredModuleList().get(0);
         List<Exam> exams = module.getExams();
         if (targetIndex.getZeroBased() >= exams.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_EXAM_DISPLAYED_INDEX);
         }
         Exam examToDelete = exams.get(targetIndex.getZeroBased());
         model.deleteExam(module, examToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_EXAM_SUCCESS, examToDelete.getName(), targetModuleCode));
+        return new CommandResult(String.format(MESSAGE_DELETE_EXAM_SUCCESS, examToDelete.getName(), module.getCode()),
+                false, GuiState.EXAMS);
```

Comment:
should be `GuiState.DETAILS`? so that can remain at the same view. if you change to `GuiState.EXAMS` then the user might be confused as to why he can't see the lessons anymore